src/main/java/graphql/nadel/engine/OverallQueryTransformer.java

```diff
@@ -84,6 +84,7 @@
         Map<String, FieldTransformation> fieldIdToTransformation = new LinkedHashMap<>();
         Map<String, String> typeRenameMappings = new LinkedHashMap<>();
         Map<String, VariableDefinition> referencedVariables = new LinkedHashMap<>();
+        List<String> hintTypenameMappings = new ArrayList<>();
```

Comment:
🤔 Deciding between a list and a map - if this remains a list, I'll change the name to `hintTypenameList`

Reply:
I think this should be encapsulated in a class (you can make up a name but perhaps something like `ExecutionRemappingState` meaning all the state that is captured to do with Nadel remapping things (renames, hydration etc...)

It can be a list or map under the covers to your hearts content.

The if statement above then becomes

```
if (... || remappingState.requiresNoMapping())
```

In fact IF you can refactor the other stateful things to be inside this class this means you can collapse a bunch of state method parameters into one class, which greatly simplifies what gets passed down